torch/ao/quantization/fx/_model_report/detector.py

```diff
@@ -0,0 +1,408 @@
+                their S metric of output of module
+                whether output of module is stationary or non-stationary
+                the tolerance level to decided whether input/output is stationary or non-stationary
+        """
+
+        # get the dictionary of the information to format the string report
+        module_dynamic_static_info = self.__generate_dict_info(model)
+
+        dynamic_vs_static_string = "Dynamic vs. Static Quantization suggestions: \n"
+
+        # This for loop goes through the information collected in module_dynamic_static_info and:
+        #   Populates the string based report with the information from module_dynamic_static_info
+        #   Compiles the complete report by appending relavent formatted strings
+
+        for module_fqn in module_dynamic_static_info.keys():
+
+            module_info = module_dynamic_static_info[module_fqn]
+            suggestion_string_template = "For module {} it is suggested to use {} quantization because {}.\n"
+
+            # decide what string formatting values will be
+            quantization_type = ""
+
+            quantization_reasoning = "the distribution of data before {} is {} and the distribution after is {}."
+            dynamic_benefit = " You will get more accurate results if you use dynamic quantization"
+            static_benefit = " You can increase model efficiency if you use static quantization"
+            benefit_str = ""
+
+            # strings for if dynamic quantized per tensor is needed
+            recommend_per_tensor = " We recommend to add a {} before this module if it is static."
+            rec_lay_to_add = "dynamic quantize per tensor layer"
+            dynamic_per_tensor_string = recommend_per_tensor.format(rec_lay_to_add)
+            dynamic_per_tensor_reasoning_string = (
+                " This is because the input to this module has a non-stationary distribution."
+            )
+
+            # start composing explanation
+            if module_info["dynamic_recommended"]:
+                quantization_type = "dynamic"
+                benefit_str = dynamic_benefit
+            else:
+                quantization_type = "static"
+                benefit_str = static_benefit
+
+            # now set the quantization explanation string
+            quantization_reasoning = (
+                quantization_reasoning.format(
+                    module_fqn, module_info["pre_observer_data_dist"], module_info["post_observer_data_dist"]
+                )
+                + benefit_str
+            )
+
+            # if we have a non-stationary input -> linear -> stationary we suggested static
+            # however, we want to also recommend they add a dynamic quantize per tensor right if this change is made
+            if (
+                module_info["pre_observer_data_dist"] == self.DEFAULT_NON_STATIONARY
+                and module_info["post_observer_data_dist"] == self.DEFAULT_STATIONARY
```

Comment:
I wonder if we should make these dict keys constant as well (like `DEFAULT_STATIONARY`) so we're not duplicating them everywhere? If there's a typo somewhere the result could be hard to debug